implementations/elixir/ockam/ockam/lib/ockam/secure_channel/channel.ex

```diff
@@ -1,380 +1,611 @@
+    field(:peer_identity_id, binary())
+    field(:h, binary())
+    field(:encrypt_st, XX.Encryptor.t())
+    field(:decrypt_st, XX.Decryptor.t())
+  end
+
+  # Secure channel' data.  Contains general fields used in every channel state, and
+  # the state' specific data (Handshaking/IdentityExchange/Established)
+  typedstruct do
+    field(:role, :initiator | :responder)
+    field(:identity, Identity.t())
+    field(:address, Ockam.Address.t())
+    field(:inner_address, Ockam.Address.t())
+    field(:vault_name, binary())
+    field(:peer_route, Ockam.Address.route())
+    field(:peer_identity_route, Ockam.Address.route())
+    field(:trust_policies, trust_policies())
+    field(:additional_metadata, map())
+    field(:channel_state, Handshaking.t() | IdentityExchange.t() | Established.t())
+  end
+
+  @handshake_timeout 20_000
+
+  @type listener_opt ::
+          {:responder_authorization, authorization()}
+          | secure_channel_opt()
+
+  @type initiator_opt ::
+          {:route, Ockam.Address.route()}
+          | secure_channel_opt()
+
+  @spec create_listener([listener_opt()]) :: {:ok, Ockam.Address.t()} | {:error, term()}
+  def create_listener(opts) do
+    Spawner.create(spawner_options(opts))
+  end
+
+  @doc """
+  Child spec to create listeners
+
+  See create_listener/1
+  """
+  @spec listener_child_spec([listener_opt()]) :: map()
+  def listener_child_spec(args) do
+    %{
+      id: __MODULE__,
+      start: {Spawner, :start_link, [spawner_options(args)]}
+    }
+  end
+
+  @spec start_link_channel([initiator_opt()], non_neg_integer) :: :ok | {:error, term()}
+  def start_link_channel(opts, handshake_timeout \\ 3000) do
+    ref = make_ref()
+
+    opts =
+      Keyword.merge(opts,
+        handshake_timeout: handshake_timeout,
+        role: :initiator,
+        waiter: {self(), ref},
+        restart_type: :temporary
+      )
+
+    {:ok, _pid, addr} = start_link(opts)
+
+    receive do
+      {:connected, ^ref} -> :ok
+    after
+      handshake_timeout ->
+        Ockam.Node.stop(addr)
+        {:error, :key_exchange_timeout}
+    end
   end
 
-  @doc false
-  @spec create(options :: Keyword.t(), timeout :: integer()) ::
-          {:ok, address :: Ockam.Address.t()}
-          | {:error, {:init_timeout, {pid :: pid(), address :: Ockam.Address.t()}}}
-          | {:error, reason :: any()}
-  def create(options, timeout \\ 30_000) when is_list(options) do
-    ## TODO: why secure channel is not a worker?
-    address_prefix = Keyword.get(options, :address_prefix, "")
-
-    options =
-      Keyword.put_new_lazy(options, :address, fn ->
-        Node.get_random_unregistered_address(address_prefix)
-      end)
-
-    case Node.start_supervised(__MODULE__, options) do
-      {:ok, pid, address} ->
-        with {:ok, _pid, address} <- wait_for_channel(pid, address, timeout) do
-          {:ok, address}
-        end
-
-      error ->
-        error
+  # Deprecated, use start_link_channel/2
```

Comment:
Optional, as it produces compilation warnings, but good for consumers' feedback IMO:
```suggestion
  @deprecated "use start_link_channel/2"
```

Reply:
Yes letting it as a comment for now.  There is code using this and the behavior is slightly different (I copied the deprecated comment from the current version)